test/test-mc-fle2-payload-iev-v2.c

```diff
@@ -0,0 +1,217 @@
+   } else {
+      ASSERT (test->type == kTypeRange);
+      // TODO (MONGOCRYPT-531) FLE2IndexedRangeEncryptedValueV2
+      ASSERT_OR_PRINT_MSG (false, "Type must be 'equality' for now");
+   }
+
+   // Validate S_KeyId as parsed.
+   const _mongocrypt_buffer_t *S_KeyId =
+      mc_FLE2IndexedEncryptedValueV2_get_S_KeyId (iev, status);
+   ASSERT_OK_STATUS (S_KeyId, status);
+   ASSERT_CMPBUF (*S_KeyId, test->S_KeyId);
+
+   // Validate bson_value_type as parsed.
+   bson_type_t bson_value_type =
+      mc_FLE2IndexedEncryptedValueV2_get_bson_value_type (iev, status);
+   ASSERT_OK_STATUS (bson_value_type, status);
+   ASSERT_CMPINT (bson_value_type, ==, test->bson_value_type);
+
+   // Decrypt ServerEncryptedValue.
+   ASSERT_OK_STATUS (mc_FLE2IndexedEncryptedValueV2_add_S_Key (
+                        crypt->crypto, iev, &test->S_Key, status),
+                     status);
+
+   // Validate K_KeyId as decrypted.
+   const _mongocrypt_buffer_t *K_KeyId =
+      mc_FLE2IndexedEncryptedValueV2_get_K_KeyId (iev, status);
+   ASSERT_OK_STATUS (K_KeyId, status);
+   ASSERT_CMPBUF (*K_KeyId, test->K_KeyId);
+
+   // Decrypt ClientEncryptedValue.
+   ASSERT_OK_STATUS (mc_FLE2IndexedEncryptedValueV2_add_K_Key (
+                        crypt->crypto, iev, &test->K_Key, status),
+                     status);
+
+   // Validate decrypted value.
+   const _mongocrypt_buffer_t *bson_value =
+      mc_FLE2IndexedEncryptedValueV2_get_ClientValue (iev, status);
+   ASSERT_OK_STATUS (bson_value, status);
+   ASSERT_CMPBUF (*bson_value, test->bson_value);
+
+   // All done!
+   mc_FLE2IndexedEncryptedValueV2_destroy (iev);
+   mongocrypt_destroy (crypt);
+   mongocrypt_status_destroy (status);
+}
+
+static void
+test_fle2_ieevv2_test (_mongocrypt_tester_t *tester, const char *path)
+{
+   printf ("Loading test from %s...\n", path);
+
+   mongocrypt_binary_t *test_bin = TEST_FILE (path);
+   if (!test_bin) {
+      TEST_ERROR ("Failed loading test data file '%s'\n", path);
+   }
+   if (test_bin->len == 5) {
+      TEST_ERROR ("Invalid JSON in file '%s'\n", path);
+   }
+
+   bson_t test_bson;
+   ASSERT (bson_init_static (&test_bson, test_bin->data, test_bin->len));
+   ASSERT (bson_validate (&test_bson, BSON_VALIDATE_NONE, NULL));
+
+   _mc_fle2_ieevv2_test test = {.payload = {0}};
+   bson_iter_t iter;
+   ASSERT (bson_iter_init (&iter, &test_bson));
+   ASSERT (_mc_fle2_ieevv2_test_parse (&test, &iter));
+   _mc_fle2_ieevv2_test_run (tester, &test);
+   _mc_fle2_ieevv2_test_destroy (&test);
+}
+
+static void
+test_fle2_ieevv2 (_mongocrypt_tester_t *tester)
+{
+   if (!_aes_ctr_is_supported_by_os) {
+      printf ("Common Crypto with no CTR support detected. Skipping.");
+      return;
+   }
+
+   // Producted by Server test: (FLECrudTest, insertOneV2)
+   test_fle2_ieevv2_test (tester,
+                          "test/data/iev-v2/FLECrudTest-insertOneV2.json");
+}
+
+void
+_mongocrypt_tester_install_fle2_ieevv2_payloads (_mongocrypt_tester_t *tester)
```

Comment:
since the plan is to use this same tester for range as well, we should get rid of the extra 'e' in `ieevv2`